### PR TITLE
#4 #1 #2 #3 Functional API 

### PR DESCRIPTION
**Pull request by dummer3**
TP2 work done

### CODE REVIEW

**Comment by SomeOne768** [comment]
Je peux ?

File: Bataille Navale.sln [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.4.33205.214
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Bataille Navale", "Bataille Navale\Bataille Navale.csproj", "{EC597957-D382-4F75-98DB-8FA53F37B457}"
```
Comment (SomeOne768) [inline Bataille Navale.sln:6]: OHohoho

File: NavalWar.DTO/Game.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+        public Map[] ListPlayer { get; } = new Map[2];
+        public int idGame { get; }
+
+        public Game(int idGame)
+        {
+            idGame = idGame;
+            ListPlayer[0] = new Map("Joueur1");
+            ListPlayer[1] = new Map("Joueur2");
+            _duration= 0;
```
Comment (antoinemeyer5) [inline NavalWar.DTO/Game.cs:17]: init `_result` & `_winnerName` missing

File: NavalWar.DTO/Game.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿namespace NavalWar.DTO
+{
+    public class Game
+    {
+        private int _result;
+        private string _winnerName;
+        private float _duration;
+
+        public Map[] ListPlayer { get; } = new Map[2];
+        public int idGame { get; }
```
Comment (antoinemeyer5) [inline NavalWar.DTO/Game.cs:10]: class attribut no ? is auto incrementing to be preferred ?

File: NavalWar.DTO/Map.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+        private HashSet<(int, int)> _listTarget = new HashSet<(int, int)>();
+
+        private string _name;
+        private Player _associatedPlayer;
+
+        public Map(string name, int lineMax, int columnMax)
+        {
+
+            Id = _rand.Next();
```
Comment (antoinemeyer5) [inline NavalWar.DTO/Map.cs:32]: very weird : id should be auto incremented no ?

File: NavalWar.DTO/Game.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+        public Game(int idGame)
+        {
+            idGame = idGame;
+            ListPlayer[0] = new Map("Joueur1");
+            ListPlayer[1] = new Map("Joueur2");
+            _duration= 0;
+        }
+
+        public Game() : this(0)
```
Comment (deeb4905) [inline NavalWar.DTO/Game.cs:20]: When we deal with several games (and therefore a game table), delete the constructor asking for the ID. Simply put one without parameters, and the ID will be the corresponding table index

File: NavalWar.DTO/Player.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NavalWar.DTO
+{
+    public class Player
+    {
+        string name;
+        List<int> history;
```
Comment (antoinemeyer5) [inline NavalWar.DTO/Player.cs:12]: here int are game id, maybe add a comment about this

**Review by antoinemeyer5 — COMMENTED** [review]
Just comment but to keep in mind and maybe create issue about this.

**Review by deeb4905 — APPROVED** [review]

**Comment by antoinemeyer5** [comment]
Very good !

**Pull request merged**